### PR TITLE
MNTOR-3261/instrument backend glean

### PR DESCRIPTION
**Pull request by rhelmer**
# References:

Jira: MNTOR-3261

# Description

Simple implementation for Glean backend, using similar API to the frontend `record()`.

NOTE: this is branched from https://github.com/mozilla/blurts-server/pull/4544 and is currently blocked from landing on it, but I think we can make progress in the meantime.

# How to test

Create or delete an account, it will be logged. In GCP environments, it will be treated as a structured log message and should be routed to the Mozilla Telemetry infra.

# Checklist (Definition of Done)

- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/telemetry/backend-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1896222
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - rhelmer@mozilla.com
+    expires: never
+    send_in_pings:
+      - technical
+
```
Comment (mansaj) [inline src/telemetry/backend-metrics.yaml:210]: Additional event we might want to track is applying coupon code to a subscription

**Review by mansaj — APPROVED** [review]
LGTM

**Comment by rhelmer** [comment]
This is going to need to be rebased against https://github.com/mozilla/blurts-server/pull/4544, we were asked to change the metrics YAML in https://github.com/mozilla/blurts-server/pull/4544#issuecomment-2171967043.

**Comment by rhelmer** [comment]
It looks like `buildGleanTypes.js` doesn't quite understand the structure of the new `backend-metrics.yaml` - I'll file a follow-up ticket to add this so we get proper types here.

There's been a request from DS/DE that we change the frontend `metrics.yaml` to match it, so we'll need it for that anyway. I'm going to land as-is which just means that we'll miss type checking for now but we can at least start testing to make sure the backend -> Glean pipeline is working.

**Pull request merged**